gremlin-python/src/main/python/gremlin_python/driver/auth.py

```diff
@@ -0,0 +1,81 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import abc
+
+
+class Auth(metaclass=abc.ABCMeta):
+
+    @abc.abstractmethod
+    def apply(self, request):
+        """Applies the necessary authentication operations to the request and returns the modified request."""
+        pass
+
+    @staticmethod
+    def basic(username, password):
+        return BasicAuth(username, password)
+
+    @staticmethod
+    def sigv4(region_name, aws_access_key_id='', aws_secret_access_key='', session_token='', service_name=''):
+        return SigV4Auth(region_name, aws_access_key_id, aws_secret_access_key, session_token, service_name)
+
+
+class BasicAuth(Auth):
+
+    def __init__(self, username, password):
+        self._username = username
+        self._password = password
+
+    def apply(self, request):
+        from aiohttp import BasicAuth as aiohttpBasicAuth
+
+        return request['headers'].update({'authorization': aiohttpBasicAuth(self._username, self._password).encode()})
+
+
+class SigV4Auth(Auth):
+
+    def __init__(self, region_name, aws_access_key_id='', aws_secret_access_key='', session_token='',
+                 service_name=''):
+        import os
+
+        self._region_name = region_name
+        self._aws_access_key_id = aws_access_key_id if aws_access_key_id else os.environ.get('AWS_ACCESS_KEY_ID')
```

Comment:
may be better to use boto to get credentials?
something like
`session = botocore.Session()
credentials = session.get_credentials()`

Reply:
Updated to using boto3.Session() based on [sigV4 examples](https://github.com/aws-samples/sigv4-signing-examples/blob/main/sdk/python/main.py)